1 Foundations/this.js

```diff
@@ -1,37 +1,160 @@
-// Definition of 'this'
-// 'this' is an object that the function is a property of.
+/** this KEYWORD
+ * 'this' keyword in JavaScript allows to decide which object should be focal when invoking a function or a method.
+ * It allows to reuse functions in different contexts or with different objects.
+ * Sometimes it is difficult to tell what the 'this' keyword is referencing to.
+ * One can tell by looking at "where the function which is using the 'this ' keyword was invoked".
+ */
+// Lets see in detail the different ways of evaluating this keyword.
 
-/* Why is 'this' important?*/
+// 1.Implicit Binding
 
+const user = {
+  name: "ritik",
+  age: "22",
+  welcome() {
+    alert(`Hello ${this.name} `);
```

Comment:
Replace the alert with logs. 
That way, the code works properly in `repl.it` and also using node via terminal.